lab-nassir/app/filter/gallery-search.js

```diff
@@ -0,0 +1,32 @@
+'use strict';
+
+module.exports = function() {
+  return function(galleries, searchName, searchDesc) {
+    if(!searchName && !searchDesc) return galleries;
+    if(searchName && !searchDesc) {
+      let fuzzyRegex = makeFuzzyRegex(searchName);
+      return galleries.filter(gallery => {
+        return fuzzyRegex.test(gallery.name.toUpperCase());
+      });
+    }
+    if(searchDesc && !searchName) {
+      let fuzzyRegex = makeFuzzyRegex(searchDesc);
+      return galleries.filter(gallery => {
+        return fuzzyRegex.test(gallery.desc.toUpperCase());
+      });
+    }
+    if(searchDesc && searchName) {
+      let fuzzyRegexDesc = makeFuzzyRegex(searchDesc);
+      let fuzzyRegexName = makeFuzzyRegex(searchName);
+      return galleries.filter(gallery => {
+        return fuzzyRegexDesc.test(gallery.desc.toUpperCase()) && fuzzyRegexName.test(gallery.name.toUpperCase());
+      });
```

Comment:
Because our generateFuzzyRegex function will return the whole string if there is no input to the filter, we don't necessarily have to have all this conditional logic. We can instead just take in the name and description search terms, filter our galleries by name, save the result of filtering that gallery array, then filter THAT array using the desc. We then return the result of filtering the galleries through both of our filters. Even if one of the filters isn't being utilized, it won't ultimately alter the array due to the way our generateFuzzyRegex function is written.